### PR TITLE
Reaction scaling

### PR DESCRIPTION
**Pull request by lass9030**
Adds reaction scaling to Teslium+uranium and Liquid Dark Matter (Which had none).

I have nothing to cash in, but if you want it, it's here.

Vortexes (LDM and Sorium) now has a max range of 10, and pull-time of 11.
Teslium+uranium now has a max range of 8.

:cl: Alonefromhell
balance: Teslium+uranium and Liquid Dark Matter's effect actually scales to their volume now.
/:cl:

### CODE REVIEW

**Comment by variableundefined** [comment]
@lass9030 Can you cite a single fix for this

**Comment by variableundefined** [comment]
Nevermind. I'll count this as free

**Comment by lass9030** [comment]
Should fix some of the issues raised in #9912

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 1]
```diff
@@ -411,9 +411,9 @@ datum/chemical_reaction/flash_powder
 
 /datum/chemical_reaction/shock_explosion/on_reaction(datum/reagents/holder, created_volume)
 	var/turf/T = get_turf(holder.my_atom)
-	for(var/mob/living/carbon/C in view(6, T))
+	for(var/mob/living/carbon/C in view(min(12, round(created_volume * 2)), T))
 		C.Beam(T,icon_state="lightning[rand(1,12)]",icon='icons/effects/effects.dmi',time=5) //What? Why are we beaming from the mob to the turf? Turf to mob generates really odd results.
-		C.electrocute_act(3.5, "electrical blast")
+		C.electrocute_act(min(7, created_volume), "electrical blast")
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: leave this alone; this intention of this is to create the absolute lowest damage amount possible without it being under the thresh-hold to not stun.

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 2]
```diff
@@ -97,9 +97,9 @@
 /datum/chemical_reaction/sorium/on_reaction(datum/reagents/holder, created_volume)
 	if(holder.has_reagent("stabilizing_agent"))
 		return
-	holder.remove_reagent("sorium", created_volume)
 	var/turf/simulated/T = get_turf(holder.my_atom)
-	goonchem_vortex(T, 1, 5, 6)
+	goonchem_vortex(T, 1, min(10, created_volume), min(12, round(created_volume * 2)))
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: mainting the ratio would be best.

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 3]
```diff
@@ -411,9 +411,9 @@ datum/chemical_reaction/flash_powder
 
 /datum/chemical_reaction/shock_explosion/on_reaction(datum/reagents/holder, created_volume)
 	var/turf/T = get_turf(holder.my_atom)
-	for(var/mob/living/carbon/C in view(6, T))
+	for(var/mob/living/carbon/C in view(min(12, round(created_volume * 2)), T))
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: having this scale clear up to 12 is probably not a good idea.

**Comment by lass9030** [comment]
Should address the comments above.

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 4]
```diff
@@ -122,9 +122,9 @@
 /datum/chemical_reaction/liquid_dark_matter/on_reaction(datum/reagents/holder, created_volume)
 	if(holder.has_reagent("stabilizing_agent"))
 		return
-	holder.remove_reagent("liquid_dark_matter", created_volume)
 	var/turf/simulated/T = get_turf(holder.my_atom)
-	goonchem_vortex(T, 0, 5, 6)
+	goonchem_vortex(T, 0, min(10, created_volume), min(11, created_volume++))
```
Comment (farie82) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: Same here. created_volume + 1

Comment (farie82) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: Same here. created_volume + 1

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 5]
```diff
@@ -135,7 +135,7 @@
 
 /datum/chemical_reaction/ldm_vortex/on_reaction(datum/reagents/holder, created_volume)
 	var/turf/simulated/T = get_turf(holder.my_atom)
-	goonchem_vortex(T, 0, 5, 6)
+	goonchem_vortex(T, 0, min(10, created_volume), min(11, created_volume++))
```
Comment (farie82) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: same

**Comment by lass9030** [comment]
Lesson learned.

**Pull request merged**